server/oc_cmds.go

```diff
@@ -43,29 +47,64 @@ func ShowValidatorCmd() *cobra.Command {
 		RunE: func(cmd *cobra.Command, args []string) error {
 			serverCtx := GetServerContextFromCmd(cmd)
 			cfg := serverCtx.Config
-
-			privValidator := pvm.LoadFilePV(cfg.PrivValidatorKeyFile(), cfg.PrivValidatorStateFile())
-			pk, err := privValidator.GetPubKey()
-			if err != nil {
-				return err
-			}
-			sdkPK, err := cryptocodec.FromOcPubKeyInterface(pk)
-			if err != nil {
-				return err
-			}
-			clientCtx := client.GetClientContextFromCmd(cmd)
-			bz, err := clientCtx.Codec.MarshalInterfaceJSON(sdkPK)
-			if err != nil {
-				return err
-			}
-			fmt.Println(string(bz))
-			return nil
+			return showValidator(cmd, cfg)
 		},
 	}
 
 	return &cmd
 }
 
+func showValidator(cmd *cobra.Command, config *cfg.Config) error {
+	var pv types.PrivValidator
+	if config.PrivValidatorListenAddr != "" {
+		chainID, err := LoadChainID(config)
+		if err != nil {
+			return err
+		}
+		serverCtx := GetServerContextFromCmd(cmd)
+		log := serverCtx.Logger
+		pv, err = node.CreateAndStartPrivValidatorSocketClient(config.PrivValidatorListenAddr, chainID, log)
+		if err != nil {
+			return err
+		}
+	} else {
```

Comment:
Why does this `showValidator` function create private key?
I think creating private key seems to go against this function rule.

Reply:
This is the same process as [Ostracon](https://github.com/line/ostracon/blob/v1.0.7/cmd/ostracon/commands/show_validator.go#L34), and when I checked with @torao, it seems that a key is created to encrypt communication between Ostracon and KMS.